server/app/services/openApi/v2/Swagger2SchemaGenerator.java

```diff
@@ -0,0 +1,347 @@
+                                                  "application_id", DefinitionType.INTEGER)
+                                              .setFormat(Format.INT32)
+                                              .build(),
+                                          Definition.builder("create_time", DefinitionType.STRING)
+                                              .build(),
+                                          Definition.builder("language", DefinitionType.STRING)
+                                              .build(),
+                                          Definition.builder("program_name", DefinitionType.STRING)
+                                              .build(),
+                                          Definition.builder(
+                                                  "program_version_id", DefinitionType.INTEGER)
+                                              .setFormat(Format.INT32)
+                                              .build(),
+                                          Definition.builder("status", DefinitionType.STRING)
+                                              .build(),
+                                          Definition.builder("submit_time", DefinitionType.STRING)
+                                              .build(),
+                                          Definition.builder(
+                                                  "submitter_email", DefinitionType.STRING)
+                                              .build()))
+                                  .build(),
+                              Definition.builder("nextPageToken", DefinitionType.STRING).build()))
+                      .build())
+              .build();
+
+      ObjectMapper mapper = Swagger2YamlMapper.getMapper();
+      return mapper.writeValueAsString(swaggerRoot);
+    } catch (RuntimeException
+        | InvalidQuestionTypeException
+        | UnsupportedQuestionTypeException
+        | JsonProcessingException ex) {
+      throw new OpenApiGenerationException(
+          OpenApiVersion.SWAGGER_V2, "Unable to generate OpenAPI schema.", ex);
+    }
+  }
+
+  /**
+   * Returns the list of schemes to allow in the swagger schema. Special case to allow http for
+   * non-prod environments for testing purposes.
+   */
+  private ImmutableList<Scheme> getSchemes() {
+    if (openApiSchemaSettings.getAllowHttpScheme()) {
+      return ImmutableList.of(Scheme.HTTP, Scheme.HTTPS);
+    }
+
+    return ImmutableList.of(Scheme.HTTPS);
+  }
+
+  /***
+   * Entry point to start building the program specific definitions for questions
+   */
+  private ImmutableList<Definition> buildApplicationDefinitions(ProgramDefinition programDefinition)
+      throws InvalidQuestionTypeException, UnsupportedQuestionTypeException {
+
+    QuestionDefinitionNode rootNode = getQuestionDefinitionRootNode(programDefinition);
+
+    return new ArrayList<>(buildApplicationDefinitions(rootNode))
```

Comment:
Question: Why does this need to be an ArrayList?

Reply:
It doesn't. Just a byproduct of refactoring. Fixed.